PaddlePaddle/PaddleVideo
# add test and predict

This PR
- add predict function, which directly calls paddle built-in predictor to predict.
- add test function

for now,  the panorama of PaddleVideo, function level.

| func| build model |backward | loss | metric(top1/5)|predict ret|
|- |:-: |:-: |:-: |:-: |:-:|
|Train|Y|Y|Y|Y||
|Val   |Y|   |Y|Y||
|Test |Y|   |  |Y||
 |Infer(not impl) |Y|   |  |  |Y|
| Predict(py)|||  |  |Y|


- Others:
  - add save_interval and output path to refine save function.
  - fix bugs when saving the best model.
  - refactoring forward to implement inferencing

![image](https://user-images.githubusercontent.com/13645332/101360588-ce4a4f80-38d8-11eb-8a6d-03a26344848c.png)

Because of predictor will sign the forward method as the total graph, and in the old design of PaddleVideo we integrate build_model and build_loss together, so the incorrect results occur, we create new interfaces (train_step, val_step, test_step) to directly call forward_train/val and indirectly call forward to decouple build_loss and build_model now

Todo:
- check shared memory problem
- benchmark
